modules/ding_adgangsplatformen/ding_adgangsplatformen.module

```diff
@@ -326,6 +326,19 @@ function ding_adgangsplatformen_login() {
   drupal_exit($authorization_url);
 }
 
+/**
+ * Implements hooK_login_provider_user.
+ *
+ * Saves the current path and sends the user to the login page.
+ *
+ */
+function ding_adgangsplatformen_login_provider_user() {
+  // We save the current page as destination.
+  $options = ['query' => ['destination' => request_uri()]];
```

Comment:
```suggestion
  $options = ['query' => drupal_get_destination()];
```

We should be able to use [`drupal_get_destination()`](https://api.drupal.org/api/drupal/includes!common.inc/function/drupal_get_destination/7.x) here to use the framework and avoid edge cases.